lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +35,112 @@ def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        self.config = Config._extract_config_content(self)
+
+        self._seed_urls = self.config.seed_urls
+        self._num_articles = self.config.total_articles
+        self._headers = self.config.headers
+        self._encoding = self.config.encoding
+        self._timeout = self.config.timeout
+        self._should_verify_certificate = self.config.should_verify_certificate
+        self._headless_mode = self.config.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as path:
+            self.config_dict = json.load(path)
+        return ConfigDTO(self.config_dict['seed_urls'],
+                         self.config_dict['total_articles_to_find_and_parse'],
+                                self.config_dict['headers'],
+                                self.config_dict['encoding'],
+                                self.config_dict['timeout'],
+                                self.config_dict['should_verify_certificate'],
+                                self.config_dict['headless_mode'])
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as path:
+            self.config_dict = json.load(path)
+        if not isinstance(self.config_dict['seed_urls'], list):
```

Comment:
rework validation to operate on top of DTO instance

Reply:
Done